core/domain/user_services_test.py

```diff
@@ -1318,6 +1329,290 @@ def test_record_user_started_state_translation_tutorial(self):
         self.assertTrue(
             user_settings.last_started_state_translation_tutorial is not None)
 
+    def test_set_user_has_viewed_lesson_info_modal_once(self):
+        auth_id = 'test_id'
+        username = 'testname'
+        user_email = 'test@email.com'
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_services.set_username(user_id, username)
+
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
+        self.assertFalse(
+            user_settings_model.user_has_viewed_lesson_info_modal_once)
+
+        user_services.set_user_has_viewed_lesson_info_modal_once(user_id)
+
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
+        self.assertTrue(
+            user_settings_model.user_has_viewed_lesson_info_modal_once)
+
+
+class UserCheckpointProgressUpdateTests(test_utils.GenericTestBase):
+    """Tests whether user checkpoint progress is updated correctly"""
+
+    EXP_ID = 'exp_id0'
+
+    SAMPLE_EXPLORATION_YAML = (
+        """author_notes: ''
+auto_tts_enabled: true
```

Comment:
Indentation is off

Reply:
Updated

Reply:
It's still off. Indent this line and the subsequent lines by 4 spaces

Reply:
I tried to do so. But YAML is sensitive about tab spaces and it gives an error if I try to do so. The similar indentation has also been used in other part of the codebase. [Example](https://github.com/oppia/oppia/blob/develop/core/domain/exp_services_test.py#L1577)